docs/design/da-pki-revocation-design.md

```diff
@@ -0,0 +1,213 @@
+   If the content is invalid, it can be edited/removed by either owner Vendor or Trustees via Update and Delete commands.
+
+   Such validation requires resolving an external URL, and this is a non-deterministic operation:
+   some nodes may resolve the URL successfully, but some nodes may not resolve it or be redirected to a different content.
+   Non-deterministic results of validation may cause a situation that on some nodes the transaction is considered as valid,
+   and on some nodes as invalid. That will break the consensus protocol and DCL won't be able to process write requests anymore.
+   An alternative option is to pass the whole content of `DataUrl` to the transaction, but it doesn't make a lot of sense because
+    1. the content can be quite big (1 MB)
+    2. there will be no guarantee that `DataUrl` actually points to the same content as passed
+
+2. Validation of `CRLSignerCertificate`. If `CRLSignerCertificate` is a PAA, then it must be present on DCL (there must be a PAA on DCL with the same pem value).
+   If `CRLSignerCertificate` is a PAI, then it must be chained back to a valid PAA present on the ledger.
+
+3. If `IssuerSubjectKeyID` is not equal to the `Subject Key Identifier` of `CRLSignerCertificate`, the transaction is considered invalid and rejected.
+
+4. If `RevocationType` is 1 (RFC5280 CRL), then `DataFileSize`, `DataDigest`, `DataDigestType` must be empty.
+
+   Otherwise, it may cause confusion. Certificates to be revoked are added to the content of CRL URL without a need to send a transaction to the ledger.
+   So, someone may look at the DCL entry, see that a file size is specified, then go to the URL, and see that the actual size there is different.
+   It may cause a question if the data is valid, and why a different file size is in DCL.
+
+5. `ProductID` is expected to be `unit16`, not `unit32` as in other places.
+
+
+## A need for Proof-of-possession of `CRLSignerCertificate` key
+No additional proof-of-possession is required because 
+- The transaction is already signed by the Vendor account key, so only a trusted Vendor can send it.
+- DCL checks that `CRLSignerCertificate` is either a valid PAI (signed by a trusted PAA from DCL) or a PAA present on the ledger.
+- It can be complicated to create an interactive PoP of `CRLSignerCertificate` for some PAIs.
+- Indirect CRLs are out of scope for now
+
+## Transactions
+
+### 1. ADD_PKI_REVOCATION_DISTRIBUTION_POINT
+Publishes a PKI Revocation distribution endpoint owned by the Vendor. 
+
+If `crlSignerCertificate` is a PAA (root certificate), then it must be present on DCL.
+
+If `crlSignerCertificate` is a PAI (intermediate certificate), then it must be chained back to a valid PAA (root certificate) present on DCL.
+In this case `crlSignerCertificate` is not required to be present on DCL, and will not be added to DCL as a result of this transaction.
+If PAI needs to be added to DCL, it should be done via `ADD_X509_CERT` transaction.
```

Comment:
I think it's not obvious (at least from the DCL API point of view) that this PAI is not added to DCL (especially since we require that PAA must be on DCL) . I propose to keep this statement to avoid further questions and confusion.